File: Shemooi/app/src/main/java/com/ecommerce/shemooi/view/MainActivity.java
package com.ecommerce.shemooi.view;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.ecommerce.shemooi.R;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
        Button continue_button = findViewById(R.id.continue_button);
        continue_button.setOnClickListener(v -> {
            startActivity(new Intent(this, MainActivity2.class));
        });
    }
}

File: Dagger2/app/src/main/java/com/example/dagger2/MainActivity.kt
package com.example.dagger2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import javax.inject.Inject

class MainActivity : AppCompatActivity() {
    @Inject
    lateinit var userRegistrationService: UserRegistrationService
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val component = DaggerUserRegistrationComponent.builder().build()
        component.getInject(this)
        userRegistrationService.registerUser("saurabh@gmail.com", "saurabh@123")
    }
}